src/gui/layout/qgslayoutview.cpp

```diff
@@ -64,7 +65,12 @@ QgsLayoutView::QgsLayoutView( QWidget *parent )
   mPreviewEffect = new QgsPreviewEffect( this );
   viewport()->setGraphicsEffect( mPreviewEffect );
 
+  connect( this, &QgsLayoutView::extentChanged, this, &QgsLayoutView::onExtentChanged );
   connect( this, &QgsLayoutView::zoomLevelChanged, this, &QgsLayoutView::invalidateCachedRenders );
+  connect( this, &QgsLayoutView::zoomLevelChanged, this, &QgsLayoutView::extentChanged );
+
+  verticalScrollBar()->installEventFilter( this );
+  horizontalScrollBar()->installEventFilter( this );
```

Comment:
Could all this logic be moved into the existing QgsLayoutView::viewChanged method? That should already be getting called whenever the view extent is changed, and already handles all the different ways this can happen.

That should also avoid the  `emit view()->extentChanged();` calls from other classes, which is rather fragile.

Reply:
The problem is `viewChanged` is called every time the scrollbar moves. So dragging the scrollbar handle will trigger many *MANY* viewChanged calls, and we dont want them to clutter the zoom level history.

Reply:
See above -- I'd handle this by using a timer to "collapse" extents which were used very close to each other.

Reply:
Any thoughts on this @YoannQDQ ?

Reply:
I agree that the approach you suggested is better (it could also be applied to the map canvas imho). There's some additional consideration in the layout view since changing the scale will sometime make the scalebars appears/disappear, which in turn slightly change the view center and does not play well with the zoom/center history. It is on my backlog, but I haven't had the time to finalize it yet.